jubatus/server/framework/mixer/push_mixer.cpp

```diff
@@ -404,19 +421,24 @@ void push_mixer::mix() {
   LOG(INFO) << (end - start) << " time elapsed "
             << s_pull << " pulled  "
             << s_push << " pushed";
-  mix_count_++;
+
+  {
+    scoped_lock lk(m_);
+    mix_count_++;
+  }
 }
 
 byte_buffer push_mixer::pull(const msgpack::object& arg_obj) {
+  scoped_rlock lk_read(model_mutex_);
+  scoped_lock lk(m_);  // Prevent `stabilizer_loop` to awake from `wait`.
+
   if (arg_obj.type != msgpack::type::RAW) {
     throw msgpack::rpc::argument_error();
   }
   msgpack::unpacked msg;
   msgpack::unpack(&msg, arg_obj.via.raw.ptr, arg_obj.via.raw.size);
   msgpack::object arg = msg.get();
 
-  scoped_rlock lk_read(model_mutex_);
```

Comment:
I think it is better to take a lock here, instead of the beginning, as msgpack unpacking can be processed simultaneously (note that multiple MIX sessions may occur simultaneously in push mixer)